set.mm

```diff
@@ -741438,6 +741624,644 @@ corresponds to the definition of an (infinite) product in [Lang] p. 5,
+    mapval efmndbas df-sn 3eqtr4i ) BBCDZAEZBFZAGZBHIZJIZBKUBBBUCLZAGUEBBAMMRUH
+    UDAUCBNOPBUGUFUFQUGQSABTUA $.
+
+  ${
+    efmnd1bas.1 $e |- G = ( EndoFMnd ` A ) $.
+    efmnd1bas.2 $e |- B = ( Base ` G ) $.
+    ${
+      efmnd1bas.0 $e |- A = { I } $.
+      $( The monoid of endofunctions on a singleton has cardinality ` 1 ` .
+         (Contributed by AV, 27-Jan-2024.) $)
+      efmnd1hash $p |- ( I e. V -> ( # ` B ) = 1 ) $=
+        ( wcel chash cfv cexp co c1 cfn wceq csn snfi ax-mp syl5eq efmndhash cz
+        eqeltri fveq2i hashsng oveq12d 1z 1exp syl6eq ) DEIZBJKZAJKZULLMZNAOIUK
+        UMPADQZOHDRUCABCFGUASUJUMNNLMZNUJULNULNLUJULUNJKNAUNJHUDDEUETZUPUFNUBIU
+        ONPUGNUHSUIT $.
+
+      $d A p $.  $d I p $.  $d V p $.
+      $( The monoid of endofunctions on a singleton is consisting of the
+         identity only.  (Contributed by AV, 31-Jan-2024.) $)
+      efmnd1bas $p |- ( I e. V -> B = { { <. I , I >. } } ) $=
+        ( vp wcel csn cmap co cop cefmnd cfv fveq2i eqtri efmndbas cv wceq fsng
+        wf wb anidms snex elmap velsn 3bitr4g eqrdv syl5eq ) DEJZBDKZUMLMZDDNKZ
+        KZUMBCCAOPUMOPFAUMOHQRGSULIUNUPULUMUMITZUCZUQUOUAZUQUNJUQUPJULURUSUDDDE
+        EUQUBUEUMUMUQDUFZUTUGIUOUHUIUJUK $.
+    $}
+
+    ${
+      efmnd2bas.0 $e |- A = { I , J } $.
+      $( The monoid of endofunctions on a (proper) pair has cardinality ` 4 ` .
+         (Contributed by AV, 18-Feb-2024.) $)
+      efmnd2hash $p |- ( ( I e. V /\ J e. W /\ I =/= J ) -> ( # ` B ) = 4 ) $=
+        ( wcel w3a chash cfv cexp co c4 cfn wceq c2 wne eqeltri efmndhash ax-mp
+        cpr prfi fveq2i cvv elex id 3anim123i hashprb syl5eq oveq12d sq2 syl6eq
+        sylib ) DFKZEGKZDEUAZLZBMNZAMNZVCOPZQARKVBVDSADEUEZRJDEUFUBABCHIUCUDVAV
+        DTTOPQVAVCTVCTOVAVCVEMNZTAVEMJUGVADUHKZEUHKZUTLVFTSURVGUSVHUTUTDFUIEGUI
+        UTUJUKDEULUQUMZVIUNUOUPUM $.
+$(
+      @( The monoid of endofunctions on a pair is consisting of the identity,
+         the transposition and two mappings to the same element.  (Contributed
+         by AV, 18-Feb-2024.) @)
+      efmnd2bas @p |- ( ( I e. V /\ J e. W )
+                       -> B = ( { { <. I , I >. , <. J , J >. } ,
+                                { <. I , J >. , <. J , I >. } } u.
+                                { { <. I , I >. , <. J , I >. } ,
+                                { <. I , J >. , <. J , J >. } } ) ) @=
+        ? @.
+$)
+    $}
+  $}
+
+  ${
+    submefmndbi.g $e |- M = ( EndoFMnd ` A ) $.
+    submefmndbi.b $e |- B = ( Base ` M ) $.
+    submefmndbi.0 $e |- .0. = ( 0g ` M ) $.
+    $( Characterization of submonoids of the monoid of endofunctions on a set
+       ` A ` .  (Contributed by AV, 17-Feb-2024.) $)
+    submefmndbi $p |- ( A e. V -> ( S e. ( SubMnd ` M )
+                       <-> ( S C_ B /\ .0. e. S /\ ( M |`s S ) e. Mnd ) ) ) $=
+      ( wcel cmnd csubmnd cfv wss cress co w3a wb efmndmnd eqid issubm2 syl ) A
+      EJDKJCDLMJCBNFCJDCOPZKJQRADEGSBCUCDFHIUCTUAUB $.
```

Comment:
Ok, but this seems like a direct application of `issubm2`, meaning this is valid for _any_ monoid.

Reply:
You are right, this theorem does not give much additional value. I will remove it and replace its (single) usage in ~idresefmnd by ~issubm2.